source/test_runner/binary_tests.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2017- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include "windows.h"
+
+#include "Catch/catch.hpp"
+
+#include "hourglass.h"
+
+namespace BinaryTests
+{
+    namespace filesystem = std::experimental::filesystem;
+
+    namespace
+    {
+        constexpr char TEMP_DIR_NAME[] = "temp";
+        constexpr char OUTPUT_LOG_FILENAME[] = "output.log";
+        constexpr char FAILED_DIR_NAME[] = "failed";
+
+        struct InputPair
+        {
+            filesystem::path input;
+            filesystem::path log;
+        };
+
+        filesystem::path gs_base_path;
+        std::map<filesystem::path, std::vector<InputPair>> gs_tests;
+
+        class TempDir
+        {
+        public:
+            TempDir()
+            {
+                filesystem::create_directory(gs_base_path / TEMP_DIR_NAME);
+            }
+
+            ~TempDir()
+            {
+                filesystem::remove_all(gs_base_path / TEMP_DIR_NAME);
+            }
+        };
+
+        void CopyContentsToTempDir(const filesystem::path& folder)
+        {
+            filesystem::copy(folder, gs_base_path / TEMP_DIR_NAME, filesystem::copy_options::recursive);
+        }
+
+        std::vector<std::wstring> ReadLog(const filesystem::path& path)
+        {
+            std::vector<std::wstring> rv;
+
+            std::wifstream file(path);
+            std::wstring line;
+            while (std::getline(file, line))
```

Comment:
I'm used to reading the file like this for some reason. I suppose `while (file >> line)` would do the same thing?